### ISSUE TITLE
client program segfault

### ISSUE THREAD

**Issue by shellcoder641** [t7hm1/pchat#1]
I tried testing the program by creating a room and have another user join, but when the other user issue the command -join room_name, the client program segfault and exit.

**Comment by st-mich43l** [comment]
did u change anythings ?

**Comment by shellcoder641** [comment]
No, I didn't. I ran the program as is. I think it's the problem with pyqt

**Comment by st-mich43l** [comment]
u might setup a new machine on virtualbox and try it again,dont try both in the same machine

**Comment by shellcoder641** [comment]
I guess so, but I have some suggestions for it:  like when a user connects to the server, the client should show some tips and commands so they know how to create a room or other things.

**Comment by shellcoder641** [comment]
Oh hey I just got a segfault again. So this is what I did: start the server first, let the 2 users connect to the server. User 1 created a room called hacking. User 2 attempted to join then both segfault, leading to the server segfault.

**Comment by st-mich43l** [comment]
> I guess so, but I have some suggestions for it: like when a user connects to the server, the client should show some tips and commands so they know how to create a room or other things.

actually i made that when user send msg when them not joined in room chat.

**Comment by st-mich43l** [comment]
> Oh hey I just got a segfault again. So this is what I did: start the server first, let the 2 users connect to the server. User 1 created a room called hacking. User 2 attempted to join then both segfault, leading to the server segfault.

please setup virtual machine dont try in the same machine

**Comment by st-mich43l** [comment]
i tested it and everything run is fine.Make sure about nickname,cause i still not add any funct to detect same nickname,second host:port of server to connect

**Comment by shellcoder641** [comment]
> > I guess so, but I have some suggestions for it: like when a user connects to the server, the client should show some tips and commands so they know how to create a room or other things.
> 
> actually i made that when user send msg when them not joined in room chat.

yeah but what if they don't send message? I think its good to just put them there when they first connected so users know what to do.

**Comment by shellcoder641** [comment]
> i tested it and everything run is fine.Make sure about nickname,cause i still not add any funct to detect same nickname,second host:port of server to connect

yeah they have different nicknames, I used the default host:port.

**Comment by st-mich43l** [comment]
> > > I guess so, but I have some suggestions for it: like when a user connects to the server, the client should show some tips and commands so they know how to create a room or other things.
> > 
> > 
> > actually i made that when user send msg when them not joined in room chat.
> 
> yeah but what if they don't send message? I think its good to just put them there when they first connected so users know what to do.

i made same to hexchat bro :D

**Comment by st-mich43l** [comment]
> > i tested it and everything run is fine.Make sure about nickname,cause i still not add any funct to detect same nickname,second host:port of server to connect
> 
> yeah they have different nicknames, I used the default host:port.

default host:port just used for when u test it in same machine,host must change to server's IP
